src/main/java/io/github/fablabsmc/fablabs/api/fluidvolume/v1/volume/ImmutableFluidVolume.java

```diff
@@ -0,0 +1,55 @@
+package io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume;
+
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.math.Fraction;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.api.FluidContainer;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.api.FluidVolume;
+
+import net.minecraft.fluid.Fluid;
+import net.minecraft.nbt.CompoundTag;
+
+/**
+ * a fluid volume who's fluid, contents, and NBT cannot change.
+ */
+public class ImmutableFluidVolume extends FluidVolume {
+	public static final FluidVolume EMPTY = new ImmutableFluidVolume();
+
+	/**
+	 * create an immutable fluid volume with no fluid (empty), no amount, and no data.
+	 */
+	public ImmutableFluidVolume() {
+	}
+
+	/**
+	 * create an immutable fluid volume.
+	 */
+	public ImmutableFluidVolume(Fluid fluid, Fraction amount, CompoundTag tag) {
+		super(fluid, amount, tag);
+	}
+
+	/**
+	 * create an immutable fluid volume with no data.
+	 */
+	public ImmutableFluidVolume(Fluid fluid, Fraction amount) {
+		super(fluid, amount);
+	}
+
+	@Override
+	public Fraction merge(FluidVolume volume) {
```

Comment:
Feels like silently failing these might cause invisible bugs. Not a hundred percent sure how good of an idea that is.

Reply:
I'm not sure what you mean, I just use the intended functionality to my advantage here

Reply:
I'm worried about folks going "I'm trying to interact with a bucket but every time I call `drain` it never gets anything!"